src/XlsxWorkSheet.cpp

```diff
@@ -14,30 +14,31 @@ IntegerVector parse_ref(std::string ref) {
 }
 
 [[cpp11::register]]
-List read_xlsx_(std::string path, int sheet_i,
-                IntegerVector limits, bool shim,
-                RObject col_names, RObject col_types,
+cpp11::list read_xlsx_(std::string path, int sheet_i,
+                cpp11::writable::doubles limits, bool shim,
+                cpp11::sexp col_names, cpp11::sexp col_types,
                 std::vector<std::string> na, bool trim_ws,
                 int guess_max = 1000, bool progress = true) {
 
   // Construct worksheet ----------------------------------------------
-  XlsxWorkSheet ws(path, sheet_i, limits, shim, progress);
+  XlsxWorkSheet ws(path, sheet_i, cpp11::as_integers(limits), shim, progress);
 
   // catches empty sheets and sheets where requested rectangle contains no data
   if (ws.nrow() == 0 && ws.ncol() == 0) {
-    return Rcpp::List(0);
+    cpp11::list x;
```

Comment:
This would need to be a writable list I think? Otherwise it is going to return `NULL` instead of `list()`
```suggestion
cpp11::writable::list();
```